walk/git.go

```diff
@@ -89,23 +122,27 @@ func (g *gitWalker) Walk(ctx context.Context, fn WalkFunc) error {
 		}
 
 		return filepath.Walk(path, func(path string, info fs.FileInfo, _ error) error {
-			if info.IsDir() {
+			// ignore directories and symlinks
+			if info.IsDir() || info.Mode()&os.ModeSymlink == os.ModeSymlink {
 				return nil
 			}
 
-			relPath, err := filepath.Rel(g.root, path)
+			relPath, err := g.relPath(path)
 			if err != nil {
-				return err
+				return fmt.Errorf("failed to determine a relative path for %s: %w", path, err)
 			}
 
-			if _, ok := cache[relPath]; !ok {
+			if entry, ok := cache[relPath]; !ok {
 				log.Debugf("path %v not found in git index, skipping", path)
 				return nil
+			} else if !hasChanged(entry, info) {
```

Comment:
Interesting that `entry` is available in this scope. I didn't know that the scope would expand to the else branch as well.